src/python/tests/core/bot/minimizer/js_minimizer_test.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the Java Script minimizer."""
+
+import unittest
+
+from bot.minimizer import js_minimizer
+from bot.tokenizer import antlr_tokenizer
+from bot.tokenizer.grammars.JavaScriptLexer import JavaScriptLexer
+from tests.test_libs import helpers
+
+
+class JSMinimizerTest(unittest.TestCase):
+  """Test for Javascript Minimizer."""
+
+  def setUp(self):
+    self._hypotheses_tested = []
+
+    helpers.patch_environ(self)
+    helpers.patch(self, [
+        'bot.minimizer.minimizer.Testcase.prepare_test',
+        'bot.minimizer.minimizer.Testcase.process'
+    ])
+
+    self.mock.prepare_test.side_effect = self._mock_prepare_test
+    self.mock.process.side_effect = self._mock_process()
+
+    self._tokenizer = antlr_tokenizer.AntlrTokenizer(JavaScriptLexer)
+    self._minimizer = js_minimizer.JSMinimizer(
+        self._mock_test_function(),
+        tokenizer=self._tokenizer.tokenize,
+        token_combiner=self._tokenizer.combine)
+
+  def _mock_process(self):
```

Comment:
Is this actually necessary since it has no behavior? Could you try running the tests without setting this and deleting it if it's unnecessary.